File: en/java-spring-demo/src/main/java/com/itangcent/springboot/demo/common/dto/UserDtoWithEnum.java
package com.itangcent.springboot.demo.common.dto;

import com.itangcent.springboot.demo.common.constant.UserType;

/**
 * User Data Transfer Object with Enum Type
 * Uses UserType enum to represent user types
 * Used for standardized transmission and processing of user types
 */
public class UserDtoWithEnum {
    /**
     * User ID
     * Unique identifier within the system
     */
    private Long id;

    /**
     * User Name
     */
    private String name;

    /**
     * User Type
     * Uses UserType enum for type safety
     */
    private UserType type;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public UserType getType() {
        return type;
    }

    public void setType(UserType type) {
        this.type = type;
    }
} 

File: cn/kotlin-spring-demo/src/main/kotlin/com/itangcent/common/dto/UserDtoWithEnum.kt
package com.itangcent.common.dto

import com.itangcent.common.constant.UserType

/**
 * 带枚举类型的用户DTO
 * 使用UserType枚举表示用户类型
 * 用于标准化用户类型的传输和处理
 */
data class UserDtoWithEnum(
    /**
     * 用户ID
     * 系统中的唯一标识符
     */
    var id: Long? = null,

    /**
     * 用户名称
     */
    var name: String? = null,

    /**
     * 用户类型
     * 使用UserType枚举类型以保证类型安全
     */
    var type: UserType? = null
) 